Planit/View/CreatePlanView.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  CreatePlanView.swift
+//  Planit
+//
+//  Created by Shivangi on 06/12/22.
+//
+
+import SwiftUI
+
+struct CreatePlanView: View {
+    
+    let user: AppUser
+    
+    @State var title: String = ""
+    @State var PlanDescription: String = ""
+    @State var PlanType: String = ""
+    @State var completed: Bool = false
+    
+    @ObservedObject var viewModel: planViewModel
+   
+    var body: some View {
+        VStack{
+            
+            PlanButtonStack(viewModel: viewModel)
+                .padding()
+                .padding(.top)
+            
+            PlanTextField(text: $title, placeholder: "Planit")
+                .padding([.horizontal, .bottom])
+            
+            CustomTextField(text: $PlanDescription, placeholder: Text("Description"),    imageName: "line.3.horizontal",
+                            foregroundColor: .gray)
+                            .padding([.horizontal, .bottom])
+            HStack{
+                Button{
+                    viewModel.uploadplan(Plan: plan(ownerUid: user.id ?? "", title: title, description: PlanDescription, planType: viewModel.filterplanSelected == .all ? "Extra" :
+                                                        viewModel.filterplanSelected.rawValue, completed: false))
+                    viewModel.showCreateplanView = false
+                    
+                }label: {
+                    CreatePlanButton()
```

Comment:
I typically see the action button on the right and the `cancel` button on the left rather than this. Another option might be to have a full-width create button as the main button, and have cancel be an x on the top of the view or something like that.